lib/configuration.js

```diff
@@ -128,14 +150,14 @@ confProto.onPush = function (pushRequest) {
         self._log.info("Received push promise for: " + pushRequest.href);
     }
 
-    var requestInfo = new RequestInfo(pushRequest.method, pushRequest.href, pushRequest.headers);
-    
     // Set requestInfo revalidate state
     // TODO pass pushRequest for future dedup pending requestInfo
-    cache.revalidate(requestInfo);
+    var requestInfo = new RequestInfo(pushRequest.method, pushRequest.href, pushRequest.headers);
 
     pushRequest.on('response', function (response) {
 
+        cache.revalidate(requestInfo);
```

Comment:
@hthetiot I think this change might be incorrect,  it should be up above (as before), because that is when the push stream opens.  

Really, the method name should probably be "revalidating",